compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/parser/ManifestBuildListener.java

```diff
@@ -170,16 +171,22 @@ private void setToManifest(TomlParser.ArrayValuesContext arrayValuesContext) {
     /**
      * Add table headers in the toml file.
      *
-     * @param key key specified in the header
+     * @param keys key specified in the header
      */
-    private void addHeader(List<String> key) {
-        currentHeader = key.get(0);
-        if (key.size() > 1) {
-            StringJoiner joiner = new StringJoiner(".");
-            for (int i = 1; i < key.size(); i++) {
-                joiner.add(key.get(i));
+    private void addHeader(List<String> keys) {
+        String header = keys.get(0);
+        // Check if the header is valid for the Ballerina.toml
+        if (ManifestHeader.valueOfLowerCase(header) == null) {
+            throw new BLangCompilerException("invalid header [" + header + "] found in Ballerina.toml");
+        } else {
```

Comment:
You can remove the else clause and just have the content without the clause.

Reply:
Remove the else clause with 5dd572e